panel/panel.py

```diff
@@ -63,15 +34,23 @@ def get_common_parent(paths):
     return common_path.rstrip('/\\')
 
 
-def create_path_dict(x):
-    abs_dict = {}
+def get_filenames(window, bids):
+    """
+    Return dict of buffer_id: file_name for all views in window.
+
+    Untitled buffers are file names are substituded by:
+    <untitled buffer_id>
+    """
+    return {
+        v.buffer_id(): v.file_name() or "<untitled {}>".format(v.buffer_id())
+        for v in window.views()
+        if v.buffer_id() in bids
+    }
+
+
+def create_path_dict(window, bids):
     base_dir = ""
-    for vid in x:
-        view = sublime.View(vid)
-        if view.file_name():
-            abs_dict[vid] = view.file_name()
-        else:
-            abs_dict[vid] = "<untitled " + str(view.buffer_id()) + ">"
+    abs_dict = get_filenames(window, bids)
```

Comment:
What does `abs_dir` mean? This is a mapping from bid to file_id or normalized_filename.  Something like this.

Reply:
Historically it’s a dictionary of absolute paths IIRC

Reply:
yes, historically `abs_` referring to absolute file paths. We should come up with more descriptive name though

Reply:
The docstring for get_filenames needs another look too: “buffers are file names are substituded“ has a typo and doesn’t make sense.

Anyway... How about bid_filenames?

I suck at this at least as much as everyone else. 😐

Reply:
`filenames_by_bid`

Reply:
Go for it 👍🏻👍🏻